core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java

```diff
@@ -107,10 +107,11 @@ public StartDepositResponse deposit(Sep10Jwt token, StartDepositRequest request)
     }
 
     StellarAssetInfo asset = requestValidator.getDepositAsset(request.getAssetCode());
-    if (request.getType() != null) {
-      requestValidator.validateTypes(
-          request.getType(), asset.getCode(), asset.getSep6().getDeposit().getMethods());
-    }
+    String fundingMethod =
+        request.getFundingMethod() != null ? request.getFundingMethod() : request.getType();
+    requestValidator.validateTypes(
```

Comment:
Does this throw an exception if `fundingMethod` is empty?

Reply:
yes